src/rpc-client.ts

```diff
@@ -1,95 +1,263 @@
+    sender: KeyPair;
+    recipient: string;
+    value: number;
+    fee?: number;
+    data?: string;
+}
+
+/**
+ * RPC Client for interacting with the Nimiq node
+ * Provides methods for blockchain queries and wallet operations
+ */
+export class RpcClient {
+    private _host: string;
+    private _port: string;
+    private _client: NimiqRPCClient;
+
+    /**
+     * Creates a new RPC client instance
+     * @param host - The hostname of the Nimiq node
+     * @param port - The port number of the Nimiq node
+     */
+    constructor(host: string, port: string) {
         this._host = host;
         this._port = port;
+        this._client = new NimiqRPCClient({
+            httpUrl: `http://${this._host}:${this._port}`,
+            wsUrl: `ws://${this._host}:${this._port}/ws`
+        });
     }
 
-    async isConnected() {
+    /**
+     * Checks if the client is connected to the node
+     * @returns Promise resolving to true if connected
+     */
+    async isConnected(): Promise<boolean> {
         try {
-            return !!await this.getBlockHeight();
+            const response = await this._client.blockchain.getBlockNumber();
+            return response !== undefined;
         } catch (e) {
             return false;
         }
     }
 
-    async getBlockHeight() {
-        return this._jsonRpcFetch('blockNumber');
+    /**
+     * Checks if consensus is established with the network
+     * @returns Promise resolving to consensus status
+     */
+    async isConsensusEstablished(): Promise<boolean> {
+        const response = await this._client.consensus.isConsensusEstablished();
+        return Boolean(response.data);
     }
 
-    async getBalance(address) {
-        return this._jsonRpcFetch('getBalance', address.toUserFriendlyAddress());
+    /**
+     * Gets the current block height
+     * @returns Promise resolving to block number
+     */
+    async getBlockHeight(): Promise<number> {
+        const response = await this._client.blockchain.getBlockNumber();
+        return Number(response.data);
     }
 
-    async getMempoolTransactions(includeTransactions = false) {
-        return this._jsonRpcFetch('mempoolContent', includeTransactions);
+    /**
+     * Gets the balance of an address
+     * @param address - The address to check
+     * @returns Promise resolving to balance in luna
+     */
+    async getBalance(address: string): Promise<number> {
+        const response = await this._client.blockchain.getAccountByAddress(address);
+        return Number(response.data.balance);
     }
 
-    async getTransactionReceipt(txHash) {
-        return this._jsonRpcFetch('getTransactionReceipt', txHash);
+    /**
+     * Gets transaction details by hash
+     * @param txHash - The transaction hash
+     * @returns Promise resolving to transaction details
+     * @throws If transaction not found or hash invalid
+     */
+    async getTransactionReceipt(txHash: string): Promise<Transaction> {
```

Comment:
```suggestion
    async getTransaction(txHash: string): Promise<Transaction> {
```
The method should be renamed to `getTransaction` because you're now in fact requesting a full transaction. A transaction receipt on the other hand is a receipt for transaction inclusion into the chain which includes metadata like the transaction hash, block hash and block height, but no details about the transaction.